query/src/test/java/org/infinispan/query/affinity/RpcCollector.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.query.affinity;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.infinispan.statetransfer.StateRequestCommand;
+import org.infinispan.statetransfer.StateResponseCommand;
+
+class RpcCollector {
+
+   private List<RpcDetail> rpcCollection = new ArrayList<>();
+
+   private Predicate<RpcDetail> stateTransferPredicate = rpc ->
+         Arrays.asList(StateRequestCommand.class, StateResponseCommand.class).contains(rpc.getCommand().getClass());
```

Comment:
Doesn't this build a new list on each invocation? While this looks very functionalicious and escape analysis can save the day I would still think of writing this as plain old ```rpc.getCommand().getClass() == X.class or rpc.getCommand().getClass() == Y.class```.